src/transformers/models/blip_2/modeling_blip_2.py

```diff
@@ -1827,9 +1827,18 @@ def generate(
         inputs_embeds = self.get_input_embeddings()(input_ids)
         inputs_embeds = torch.cat([language_model_inputs, inputs_embeds.to(language_model_inputs.device)], dim=1)
 
+        if not self.language_model.config.is_encoder_decoder:
+            max_length = generate_kwargs.get("max_length", 20) + language_model_inputs.shape[1]
+            min_length = min(0, generate_kwargs.get("min_length", 0) + language_model_inputs.shape[1])
+        else:
```

Comment:
I think the else is not needed :)

Reply:
> Also, I was thinking if BLIP actually should not have custom generate and can be written same was as Llava, to reduce any future bugs. It is just another possible solution, wdyt?

Maybe, but let's not spend time there :D

Reply:
btw, by 

> I would say no

I understood "no, it is not needed" [and thus should be removed]

Reply:
Sorry if I was not clear. I meant, "no, we need it for proper generation"

Reply:
I just realized why it was not needed :) okay, removed it and checked everyhtin is correct